plan/logical_plan_builder.go

```diff
@@ -660,15 +660,10 @@ func (b *planBuilder) buildProjection4Union(u *LogicalUnionAll) {
 		if _, isProj := child.(*LogicalProjection); needProjection || !isProj {
 			b.optFlag |= flagEliminateProjection
 			proj := LogicalProjection{Exprs: exprs}.init(b.ctx)
-			if childID == 0 {
```

Comment:
Why remove this?

Reply:
THe logic of generating new columns is changed. We don't need take the first children as a special one here

Reply:
We generated new columns above here. So don't need add this check.